sqlglot/dialects/snowflake.py

```diff
@@ -344,7 +344,16 @@ def _parse_table_parts(self, schema: bool = False) -> exp.Table:
                 table = self._parse_string()
 
             if table:
-                return self.expression(exp.Table, this=table)
+                file_format = None
+                pattern = None
+
+                if self._match_text_seq("(", "FILE_FORMAT", "=>"):
```

Comment:
does this syntax exist outside of this file format stuff? is it a generic kwarg for snowfake?

Reply:
Good call, I'm not sure but it seems like we have another couple of examples that look similar. It seems like this syntax is used only in a few clauses, though.

Reply:
can you parse a wrapped csv of kwargs?

Reply:
We need lookahead to ensure there's a `=>` here because you may get a table name followed by a table alias with named columns, as in

```sql
SELECT * FROM @foo t(bla)  -- valid snowflake
```

The `t(bla)` part will be consumed by `self._parse_wrapped_csv(self._parse_lambda)` as a function call instead of a table alias, so we'd need to be a little more smart than that here. Do you see any other simple way to reuse it?

I'm personally fine with this approach for now, if it's a common snowflake pattern for kwargs (i.e. we get another similar issue) we can refactor this into a helper?